### PR TITLE
gRPC Java S2A protos

### PR DESCRIPTION
**Pull request by rmehta19**
Context:
- https://github.com/grpc/grpc-java/pull/11113
- https://github.com/google/s2a-go/blob/main/README.md

### CODE REVIEW

**Comment by matthewstevenson88** [comment]
Thanks for the PR! This change LGTM. Please fix the CLA issue, remove the "draft" piece from the title, and in the PR description can we link to https://github.com/google/s2a-go/blob/main/README.md ?

**Comment by rmehta19** [comment]
@matthewstevenson88 Thanks for the review! Done.

**Comment by matthewstevenson88** [comment]
LGTM.

Please get approval from @ejona86 as well. (I think I technically have the permission to approve this, but I'm not one of the normal maintainers of this repo so I'd like to get Eric's sign-off that he is OK with where we put the protos.)

File: grpc/gcp/s2a/common.proto
Comment (ejona86) [inline grpc/gcp/s2a/common.proto:1]: Is this right? This project typically has gRPC Authors with the copyright. With Google as the copyright holder it is sort of like "why is this in the grpc package?"

Comment (rmehta19) [reply grpc/gcp/s2a/common.proto]: Updated the files to use gRPC Authors with the copyright.

File: grpc/gcp/s2a/common.proto
Comment (ejona86) [inline grpc/gcp/s2a/common.proto:20]: Should these packages be grpc.gcp.s2a? Note that is a breaking change for S2AService.

Comment (rmehta19) [reply grpc/gcp/s2a/common.proto]: Yes they should be -- thank you for catching this! I made the change in the latest commit for this PR, and the related changes have been made in the latest commit of the Java client PR: https://github.com/grpc/grpc-java/pull/11113.

**Review by ejona86 — APPROVED** [review]

**Pull request merged**
